### ISSUE TITLE
Export WinstonLogger from npm module

### ISSUE THREAD

**Issue by kierans** [gremo/nest-winston#44]
Firstly great job on this (npm) module!! It's saved me some hassle so thanks 👍 

However using the `WinstonLogger` as a `@Module` means that you have to wait for the logger to be instantiated before it will kick in which means you get the default Nest Logger's output for a bit eg:

```
[Nest] 13486   - 09/28/2019, 5:16:50 PM   [InstanceLoader] TypeOrmCoreModule dependencies initialized +1314ms
info: 9/28/2019, 5:16:50 PM	 [RoutesResolver] AppController {/}:
```

However in the [Nestjs Logger Docs](https://docs.nestjs.com/techniques/logger) it suggests passing the logger to app at app creation time eg:

```javascript
const app = await NestFactory.create(AppModule, {
  logger: createWinstonLogger({
    level: "info"
  })
});

function createWinstonLogger(opts: LoggerOptions): LoggerService {
  const logger: winston.Logger = winston.createLogger(opts);

  return new WinstonLogger(logger);
}
``` 

This leads to uniform log output ie:

```
info: 9/28/2019, 5:24:14 PM	 [InstanceLoader] TypeOrmCoreModule dependencies initialized
info: 9/28/2019, 5:24:14 PM	 [RoutesResolver] AppController {/}:
```

However because the `WinstonLogger` class is not exported, the only way to make this work for me to copy/paste it into my codebase.

**Comment by gremo** [comment]
If I understand you request, you want to manually create a instance of `WinstonLogger`? This way, any of the module options will be of course ignored... this seems to me completely wrong.

**Comment by kierans** [comment]
@gremo I don't think you've understood my use case. If you read the [NestJS documentation](https://docs.nestjs.com/techniques/logger) creating the logger outside of the DI container is a recommended pattern as it means the same logger configuration can be used by Nest (as a global logger) as well as the application code (loggers are tricky beats that straddle the application/infrastructure divide).

What is being passed to the `createWinstonLogger` function above is the same outcome as if you used `@Module` settings. I'm still creating the logger in an accepted pattern so it's not wrong.

However I need the `WinstonLogger` class to be exported from this library so that I don't have to copy/paste the code into my codebase.

Can this please be reopened/actioned?

**Comment by gremo** [comment]
> What is being passed to the createWinstonLogger function above is the same outcome as if you used @Module settings. 

How so? If we create the instance "manually" (using `return new WinstonLogger(logger);`) how the module options will be inherited?

**Comment by sjmittal** [comment]
The docs says that we can use winston logger across whole app including bootstrap by using:

import { WINSTON_MODULE_NEST_PROVIDER } from 'nest-winston';

```
async function bootstrap() {
  const app = await NestFactory.create(AppModule);
  app.useLogger(app.get(WINSTON_MODULE_NEST_PROVIDER));
}
bootstrap();
```

Isn't that same as what we are trying to do here by passing WinstonLogger manually.

**Comment by gremo** [comment]
Really confused. What do you want to export/add into this repository? Can you please explain me with real code? I will be glad to help and push the fix or new feature.

**Comment by kierans** [comment]
@sjmittal If you look at the example I gave when I opened the issue, the formatting of the log messages is different doing what you suggested. Why? It's because when the logger bean is being created as part of the application context, Nest will use it's default internal logger until the logger Module is finished being instantiated.

What I want to achieve @gremo is uniformity of output. That means the logger has to be created outside of the application modules lifecycle and injected into the application via the `logger` property which is what I do in my example when I created the ticket.

**Comment by sjmittal** [comment]
@kierans 
Yes some messages have default nest formatting, but if you read https://docs.nestjs.com/techniques/logger it says that when you add a custom logger 
> your first initialization messages won't be handled by your logger instance, though, it shouldn't really matter at this point
https://docs.nestjs.com/techniques/logger

I am not sure the PR created addresses this issue.

**Comment by kierans** [comment]
@sjmittal I disagree with the statement that my first initialisation messages not being handled shouldn't matter. That's an opinion expressed by the Nest authors. During "initialisation" (or bootstrapping) is when, for example, TypeORM tries to connect to the DB. I personally would like to know if that fails.

When I'm attaching alarms, or wanting to take other actions based on log output (eg: in a cloud based environment where a lot of things happen based on log messages) I want a consistent format for my messages to be in.

As for the PR, it's a reduction of code I've already written for my application and does what I want. So I'm not sure why you don't think it will address this issue?

**Comment by kierans** [comment]
Fixed by #62
